### ISSUE TITLE
Updating a recurring event instance operation never completes

### ISSUE THREAD

**Issue by arseni-radziuk** [microsoftgraph/msgraph-sdk-java#2170]
### Describe the bug

I am trying to modify an instance or a recurring event but this operation fails.

### Expected behavior

I should be able to modify a recurring event instance as this operation worked successfully in the previous versions of the library.

### How to reproduce

1. Create a recurring events with any pattern (e.g., weekly event occurring on Thursday and Friday starting today and with end date in 2 months).
2. Query occurring event and then query a list of its instances.
3. Take any of the queried instances and try to modify it (e.g., by setting a new subject). The operation is never completed. It looks like this operation causes an infinite loop in the library.

### SDK Version

6.16.0

### Known Workarounds

As I workaround I am able to delete the occurrence instance and then create another event.

### Debug output

<details><summary>Click to expand log</summary>
```

<Log output here>

```
</details>

**Comment by Ndiritu** [comment]
Hi @arseni-radziuk, thank you for reporting this issue.
Could you please share the failing code sample?

**Comment by arseni-radziuk** [comment]
Hi @Ndiritu, please see the code sample below.
```
       public List<Event> getInstanceEvents(String resourceId, Event masterEvent) {
        List<Event> instanceEvents = new ArrayList<>();
        LocalDateTime startDateParam = LocalDateTime.now();
        LocalDateTime endDateParam = LocalDateTime.now().plusMonths(2);
        EventCollectionResponse eventCollectionPage;
        do {
            eventCollectionPage = calendarClient.getGraphClient().users().byUserId(resourceId)
                    .events().byEventId(masterEvent.getId()).instances().get(getRequestConfiguration -> {
                        getRequestConfiguration.queryParameters.startDateTime = startDateParam.toString();
                        getRequestConfiguration.queryParameters.endDateTime = endDateParam.toString();
                        getRequestConfiguration.queryParameters.top = 100;
                        getRequestConfiguration.queryParameters.skip = instanceEvents.size();
                    });
            instanceEvents.addAll(eventCollectionPage.getValue());
        } while (CollectionUtils.isNotEmpty(eventCollectionPage.getValue()));
        instanceEvents.sort(Comparator.comparing(GraphApiUtils::getStartDate));
        return instanceEvents;
    }    
```
Then on the retrieved instances the following call is made:
`    instances.get(3).setSubject("test");`

I was only able to reproduce the issue with weekly pattern occurring on Thursday and Friday starting today and with end date in 2 months (as e.g. with daily pattern it worked fine for me).

Also, I can see the following stack trace caused by subject update operation:
![image](https://github.com/user-attachments/assets/20425397-4990-4eb5-889f-4b0d5455253d)

**Comment by Ndiritu** [comment]
Thank you for sharing this @arseni-radziuk.
This should be resolved in the latest version of this SDK `6.17.0` (release in progress). Kindly upgrade and let us know.

**Comment by arseni-radziuk** [comment]
@Ndiritu The bug is resolved for my case, thank you.
